File: abLib/abBase64Bitmap/Encoder/bin2b64.py
#!/usr/bin/python

import sys
import base64

n = len( sys.argv )
if n < 2 :
       print( "\nEncode a binary file in Base64 and return a wxBasic script with a string" )
       print( "variable ready to be used as argument in 'abBase64Bitmap' constructor.\n" )
       print( "Usage:  python bin2b64.py <binfilename> [<b64filename>]\n" )
       print( "   <binfilename>   Name of the binary file to be encoded as Base64." )
       print( "   <b64filename>   Name of the output wxBasic script with the Base64" )
       print( "                   encoded string variable. If omitted, uses" )
       print( "                   '<bin_filename>.wxb'.\n" )
       sys.exit( 0 )

bin_file = sys.argv[1]
b64_file = sys.argv[n-1]
if n < 3 : b64_file = bin_file + ".wxb"

with open( bin_file, "rb" ) as f:
        bin_lst = f.read()

mime_base64 = base64.encodebytes( bin_lst ).decode( "ascii" )
mime_base64 = mime_base64.replace( "\n", "\" &\n\t\"" )
mime_base64 = "myVar = \"" + mime_base64 + "\""

with open( b64_file, "w" ) as f:
    f.write( mime_base64 )

print( "Done.\n" )

File: abLib/abBase64Bitmap/Encoder/bin2b64.pl
#!/usr/bin/perl

use MIME::Base64;

my $n = @ARGV;
if( $n < 1 ) {
      print "\nEncode a binary file in Base64 and return a wxBasic script with a string\n";
      print "variable ready to be used as argument in 'abBase64Bitmap' constructor.\n\n";
      print "Usage:  perl bin2b64.pl <binfilename> [<b64filename>]\n\n";
      print "   <binfilename>   Name of the binary file to be encoded as Base64.\n";
      print "   <b64filename>   Name of the output wxBasic script with the Base64\n";
      print "                   encoded string variable. If omitted, uses\n";
      print "                   '<bin_filename>.wxb'.\n\n";
      exit;  }

my $bin_file = $ARGV[0];
my $b64_file = $ARGV[$n-1];
$b64_file = $bin_file . ".wxb" if( $n < 2 );

open( FILE, $bin_file );
    binmode( FILE );
    my $bin_lst = join( '', <FILE> );
close( FILE );

my $mime_base64 = encode_base64( $bin_lst );
$mime_base64 =~ s/\n/" &\n\t"/g;
$mime_base64 = "myVar = \"" . $mime_base64 . "\"";

open( FILE, ">$b64_file" );
    print FILE $mime_base64;
close( FILE );

print( "Done.\n\n" );

